### PR TITLE
[capture] Add trigger source command

### PR DESCRIPTION
**Pull request by nasahlpa**
This PR adds the SCA trigger source command to the capture scripts. Needed to start penetration tests at ASIC as there is no HW trigger available.

### CODE REVIEW

File: target/communication/sca_trigger_commands.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        Args:
+            trigger_source:
+                            - 0: Precise, hardware-generated trigger - FPGA only.
+                            - 1: Fully software-controlled trigger.
+        """
+        if self.simple_serial:
+            self.target.write(cmd="t", data=bytearray([trigger_source]))
+        else:
+            # TODO: without the delay, the device uJSON command handler program
+            # does not recognize the commands. Tracked in issue #256.
```
Comment (vrozic) [inline target/communication/sca_trigger_commands.py:32]: nit: We usually use this format for TODO. Makes it easier to search the repository.

`TODO(#256): Without the delay, the device uJSON command handler program does not recognize the commands.`

**Comment by nasahlpa** [comment]
The uJSON tests are currently failing as this PR needs the changes from #269 and #304.

File: target/communication/sca_trigger_commands.py (hunk 1 above)
Comment (vogelpi) [reply target/communication/sca_trigger_commands.py:32]: Good point raised by @vrozic. Please make sure to align the format before merging.

File: target/communication/sca_trigger_commands.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            time.sleep(0.01)
+            self.target.write(json.dumps("TriggerSca").encode("ascii"))
+            # DisableMasking command.
```
Comment (vogelpi) [inline target/communication/sca_trigger_commands.py:35]: This comment seems wrong.

**Review by vogelpi — COMMENTED** [review]
Looks good module two comments, thanks @nasahlpa !

File: target/communication/sca_trigger_commands.py (hunk 2 above)
Comment (nasahlpa) [reply target/communication/sca_trigger_commands.py:35]: Fixed, thanks!

File: target/communication/sca_trigger_commands.py (hunk 1 above)
Comment (nasahlpa) [reply target/communication/sca_trigger_commands.py:32]: Hm I followed the recommendation of the OpenTitan [C style guide](https://opentitan.org/book/doc/contributing/style_guides/c_cpp_coding_style.html#todo-comments) regarding comments. There seems not to be a recommendation for Python though.

But I am following @vrozic advise and change it to `TODO(#256)` for now and I created an issue that we can streamline this later on.

**Pull request merged**